### PR TITLE
fix CI, update php and db versions

### PR DESCRIPTION
**Pull request by wkloucek**
bit rot came faster than expected...  https://drone.owncloud.com/owncloud/qnap/146

What was done to make it work again:
- use drone config from announcementcenter (was already updated to use newer php)
- use makefile from files_external_s3 + additional l10n config
- set up sonarcloud
- test against a recent version of postgresql since we are using this in production and not a ridiculous old version
- suppress some new php lint warnings
- format the starlark file (diff becomes more readable if you skip this commit: https://github.com/owncloud/qnap/pull/41/files/4ba60bed89a621561053a0ff4e3cef812a0e9002..a3589ad00a197a1579b26a9b7a34fbfc44d3d38e#diff-6a968d4b7901c8526c8d9a86d1f434737dbfe2b00da21a7d3041ed4e3d58d62eR2)

What was not done: 
- code changes, because QA was already done and there was no intend to change anything before release. But no working CI for a release is also not nice...

### CODE REVIEW

**Comment by wkloucek** [comment]
merging this only after the release makes probably sense since we build the singed app package manually in either case...

File: Makefile [hunk 1]
```diff
@@ -95,9 +114,30 @@ test-php-phpstan: ## Run phpstan
...
 
-.PHONY: test-js
-test-js: $(nodejs_deps)
-	$(KARMA) start tests/js/karma.config.js --single-run
+.PHONY: test-acceptance-api
+test-acceptance-api: ## Run API acceptance tests
+test-acceptance-api: $(acceptance_test_deps)
+	BEHAT_BIN=$(BEHAT_BIN) ../../tests/acceptance/run.sh --remote --type api
+
+.PHONY: test-acceptance-cli
```
Comment (phil-davis) [inline Makefile:122]: some of these acceptance test targets won't be used - but it doesn't hurt to have them ready.

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Pull Requests
+sonar.pullrequest.provider=GitHub
+sonar.pullrequest.github.repository=owncloud/qnap
+sonar.pullrequest.base=${env.SONAR_PULL_REQUEST_BASE}
+sonar.pullrequest.branch=${env.SONAR_PULL_REQUEST_BRANCH}
+sonar.pullrequest.key=${env.SONAR_PULL_REQUEST_KEY}
+
+# Properties specific to language plugins:
+sonar.php.coverage.reportPaths=results/clover-phpunit-php7.3-sqlite.xml,results/clover-phpunit-php7.3-mariadb10.2.xml,results/clover-phpunit-php7.3-postgres13.3-sqlite.xml,results/clover-phpunit-php7.4-sqlite.xml,results/clover-phpunit-php7.4-mariadb10.2.xml
```
Comment (phil-davis) [inline sonar-project.properties:32]:
```suggestion
sonar.php.coverage.reportPaths=results/clover-phpunit-php7.3-sqlite.xml,results/clover-phpunit-php7.3-mariadb10.2.xml,results/clover-phpunit-php7.3-postgres13.3-sqlite.xml
```

`.drone.star` has coverage false for PHP 7.4 unit tests pipelines. So the 7.4 coverage files will not exist.

(strangely, Sonar Cloud processing does not complain about `reportPaths` that do not exist)

**Review by phil-davis — APPROVED** [review]
LGTM - I only found one little thing. Maybe adjust that and squash?

**Review by phil-davis — APPROVED** [review]

**Pull request merged**